examples/vertx/src/main/java/io/cloudevents/examples/vertx/SampleHTTPServer.java

```diff
@@ -0,0 +1,52 @@
+package io.cloudevents.examples.vertx;
+
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.http.vertx.VertxMessageFactory;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import java.util.Arrays;
+
+public class SampleHTTPServer {
+
+    public static void main(String[] args) {
+
+        if (args.length != 1) {
+            System.err.println("Usage: SampleHTTPServer <port>");
+            return;
+        }
+        final int port = Integer.parseInt(args[0]);
+
+        final Vertx vertx = Vertx.vertx();
+
+        final Promise<HttpServer> serverStartedPromise = Promise.promise();
```

Comment:
Can you avoid doing the `Promise` dance here? I think it makes the sample more complex to understand, just add the callback to listen and i think it's fine